src/dispatch/batch_builder.rs

```diff
@@ -0,0 +1,150 @@
+/// can run in parallel with other Systems.
+/// In addition the Sub Systems can run in parallel within the Batch.
+#[derive(Default)]
+pub struct BatchBuilder<'a, 'b> {
+    dispatcher_builder: DispatcherBuilder<'a, 'b>,
+}
+
+impl<'a, 'b> BatchBuilder<'a, 'b> {
+    /// Create a Batch using a dispatcher builder
+    pub fn new(dispatcher_builder: DispatcherBuilder<'a, 'b>) -> Self {
+        BatchBuilder { dispatcher_builder }
+    }
+}
+
+impl<'a, 'b> BatchBuilder<'a, 'b> {
+    /// Build the BatchSystem that will dispatch the inner dispatcher.
+    ///
+    /// Add the returned system to any dispatcher.
+    pub fn build<T>(self) -> T
+    where
+        T: System<'a> + BatchController<'a, 'b> + Send + 'a,
+    {
+        let mut reads = self.dispatcher_builder.stages_builder.fetch_all_reads();
+        reads.extend( <T::BatchSystemData as SystemData>::reads() );
+        reads.sort();
+        reads.dedup();
+
+        let mut writes = self.dispatcher_builder.stages_builder.fetch_all_writes();
+        writes.extend(<T::BatchSystemData as SystemData>::reads());
+        writes.sort();
+        writes.dedup();
+        
+        let accessor = BatchAccessor::new(reads, writes);
+        let dispatcher = self.dispatcher_builder.build();
+
+        T::create(accessor, dispatcher)
+    }
+}
+
+/// The BatchAccessor is responsible for the notification of the read and write resources
+/// of the subsystems
+pub struct BatchAccessor {
+    reads: Vec<ResourceId>,
+    writes: Vec<ResourceId>,
+}
+
+impl BatchAccessor {
+    /// Create the BatchAccessor
+    pub fn new(reads: Vec<ResourceId>, writes: Vec<ResourceId>) -> Self {
+        BatchAccessor { reads, writes }
+    }
+}
+
+impl Accessor for BatchAccessor {
+    fn try_new() -> Option<Self> {
+        None
+    }
+
+    fn reads(&self) -> Vec<ResourceId> {
+        self.reads.clone()
+    }
+
+    fn writes(&self) -> Vec<ResourceId> {
+        self.writes.clone()
+    }
+}
+
+/// The BatchUncheckedWorld wrap an instance of the world.
+/// It's safe to be used only in the context of the Batch.
+pub struct BatchUncheckedWorld<'a>(pub &'a World);
+
+impl<'a> DynamicSystemData<'a> for BatchUncheckedWorld<'a> {
+    type Accessor = BatchAccessor;
+
+    fn setup(_accessor: &Self::Accessor, _world: &mut World) {}
+
+    fn fetch(_access: &Self::Accessor, world: &'a World) -> Self {
+        BatchUncheckedWorld(world)
+    }
+}
+
+/// The BatchController is the additional trait that a normal System must implement
+/// in order to be used as BatchControllerSystem.
+pub trait BatchController<'a, 'b> {
+    
+    /// Here you must set all the Resources type that you want to use inside the
+    /// BatchControllertSystem.
+    /// All other are fetched automatically.
+    type BatchSystemData: SystemData<'a>;
+    
+    /// Create the instance of the BatchControllerSystem
+    fn create(accessor: BatchAccessor, dispatcher: Dispatcher<'a, 'b>) -> Self;
```

Comment:
Any reason why this is called `create` instead of `new`?

Reply:
Usually `new` is implemented by a struct to initialize itself so I thought was better differentiate it to make it more clear.
Do you think that new is better?